R/LSTdecisionTree.R

```diff
@@ -0,0 +1,278 @@
+                                      "Parametric test",
+                                      "Non-parametric alternative"))
+  
+  # Divide nodes into "answers to questions" and "test outcomes" to format them differently
+  answerNodesDf <- head(nodesDf, 36)
+  testNodesDf <- tail(nodesDf, 28)
+  testChoice <- tail(activeNodes, 1)
+  labelDf <- data.frame(x = 1, y = 1, label = testChoice)
+  
+  # create test outcome plot
+  plot1 <- createJaspPlot(title = gettext("Test choice"), width = 450, height = 150)
+  plotObject <- ggplot2::ggplot() +
+    ggplot2::geom_label(data = labelDf, mapping = ggplot2::aes(x = x, y = y, label = label), size = 10, fill = "#8dc63f") +
+    ggplot2::theme_void()
+  plot1$plotObject <- plotObject
+  
+  jaspResults[["DecisionTree"]] <- createJaspContainer("")
+  jaspResults[["DecisionTree"]][["Heading"]] <- plot1
+  jaspResults[["DecisionTree"]][["Text"]] <- createJaspHtml("Some explanation and maybe a link to the test?", "p")
+  
+  # create decision tree plot
+  if (options[["displayFullTree"]]) {
+    plot2 <- createJaspPlot(title = gettext("Decision Tree"), width = 2300, height = 1500)
+    plotObject <- ggplot2::ggplot() +
+      ggplot2::scale_x_reverse() +
+      ggplot2::scale_y_reverse() +
+      ggplot2::geom_rect(data = nodesDf, mapping = ggplot2::aes(xmin = xmin, ymin = ymin,
+                                                                xmax = xmax, ymax = ymax, fill = active), color = "black") +
+      ggplot2::geom_text(data = answerNodesDf, mapping = ggplot2::aes(x = x, y = y, label = label), size = 8) +
+      ggplot2::geom_text(data = testNodesDf, mapping = ggplot2::aes(x = x, y = y, label = label), size = 5) +
+      ggplot2::geom_text(data = questionsDf, mapping = ggplot2::aes(x = x, y = y, label = label), size = 7, fontface = "bold") +
+      ggplot2::geom_path(data = edgesDf, mapping = ggplot2::aes(x = x, y = y, group = id),
+                         arrow = ggplot2::arrow(length = ggplot2::unit(0.3, "cm"), type = "closed"),
+                         size = 1) +
+      ggplot2::scale_fill_manual(values = c("TRUE" = "#8dc63f", "FALSE" = "#15a1e2")) +
+      ggplot2::theme_void() +
+      ggplot2::theme(legend.position = "none")
+    plot2$plotObject <- plotObject
+    
+    jaspResults[["DecisionTree"]][["Plot"]] <- plot2
+  }
+  return()
+}
+
+.getSelectedOptions <- function(options) { # extra if conditions to prevent qml bug from selecting impossible options
+  selectedOptionsVector <- c()
+  
+  #Q1: "How many outcome variables?")
+  if (options[["nOutcomeVariables"]] == "nOutcomeOne")
+    selectedOptionsVector[1] <- "One"
+  if (options[["nOutcomeVariables"]] == "nOutcomeMany")
+    selectedOptionsVector[1] <- "Two or more"
+  
+  #Q2: "What type of outcome?"
+  if (options[["typeOutcomeVariables"]] == "typeOutcomeCont")
+    selectedOptionsVector[2] <- "Continuous"
+  if (options[["typeOutcomeVariables"]] == "typeOutcomeCat" & options[["nOutcomeVariables"]] == "nOutcomeOne") 
```

Comment:
```suggestion
  if (options[["typeOutcomeVariables"]] == "typeOutcomeCat" && options[["nOutcomeVariables"]] == "nOutcomeOne") 
```
same goes for the if statements below (also change `|` into `||`).